Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SofQWMomentsScan.py

```diff
@@ -0,0 +1,485 @@
+        scan_alg.setProperty('BackgroundRange', self._background_range)
+        scan_alg.setProperty('RebinString', self._rebin_string)
+        scan_alg.setProperty('DetailedBalance', self._detailed_balance)
+        scan_alg.setProperty('ScaleFactor', self._scale_factor)
+        scan_alg.setProperty('FoldMultipleFrames', self._fold_multiple_frames)
+        scan_alg.setProperty('GroupingMethod', self._grouping_method)
+        scan_alg.setProperty('GroupingWorkspace', self._grouping_ws)
+        scan_alg.setProperty('MapFile', self._grouping_map_file)
+        scan_alg.setProperty('UnitX', self._output_x_units)
+        scan_alg.setProperty('OutputWorkspace', self._red_ws)
+        scan_alg.execute()
+        logger.information('ReducedWorkspace : %s' % self._red_ws)
+
+        Rebin(InputWorkspace=self._red_ws,
+              OutputWorkspace=self._red_ws,
+              Params=self._energy_range,
+              EnableLogging=False)
+
+        input_workspace_names = mtd[self._red_ws].getNames()
+
+        inst = mtd[input_workspace_names[0]].getInstrument()
+        if inst.hasParameter('analyser'):
+            analyser_name = inst.getStringParameter('analyser')[0]
+            analyser_comp = inst.getComponentByName(analyser_name)
+            if analyser_comp is not None and analyser_comp.hasParameter('resolution'):
+                self._resolution = float(analyser_comp.getNumberParameter('resolution')[0])
+            else:
+                self._resolution = 0.01
+        logger.information('Resolution = %d' % self._resolution)
+
+        output_workspaces = list()
+        temperatures = list()
+        run_numbers = list()
+        sofqw_alg = self.createChildAlgorithm("SofQW", enableLogging=False)
+        group_alg = self.createChildAlgorithm("GroupWorkspaces", enableLogging=False)
+
+        for input_ws in input_workspace_names:
+            progress.report('SofQW for workspace: %s' % input_ws)
+            sofqw_alg.setProperty("InputWorkspace", input_ws)
+            sofqw_alg.setProperty("QAxisBinning", self._q_range)
+            sofqw_alg.setProperty("EMode", 'Indirect')
+            sofqw_alg.setProperty("ReplaceNaNs", True)
+            sofqw_alg.setProperty("Method", 'Polygon')
+            sofqw_alg.setProperty("OutputWorkspace", input_ws + '_sqw')
+            sofqw_alg.execute()
+            mtd.addOrReplace(input_ws + '_sqw', sofqw_alg.getProperty("OutputWorkspace").value)
+            output_workspaces.append(input_ws + '_sqw')
+
+            # Get the sample temperature
+            temp = self._get_temperature(input_ws + '_sqw')
+            if temp is not None:
+                temperatures.append(temp)
+            else:
+                # Get the run number
+                run_no = self._get_InstrRun(input_ws)[1]
+                run_numbers.append(run_no)
+
+        y_axis = mtd[input_workspace_names[0] + '_sqw'].getAxis(1)
+        y_values = y_axis.extractValues()
+        q = list()
```

Comment:
q is not a descriptive name (the only thing I can think of is charge, in which case a better variable name would be charge).